lib/connection/connection_config.js

```diff
@@ -212,63 +212,16 @@ function ConnectionConfig(options, validateCredentials, qaMode, clientInfo) {
   // if we're running in node and some proxy information is specified
   let proxy;
   if (Util.isNode() && (Util.exists(proxyHost) || Util.exists(proxyPort))) {
-    // check for missing proxyHost
-    Errors.checkArgumentExists(Util.exists(proxyHost),
-      ErrorCodes.ERR_CONN_CREATE_MISSING_PROXY_HOST);
-
-    // check for invalid proxyHost
-    Errors.checkArgumentValid(Util.isString(proxyHost),
-      ErrorCodes.ERR_CONN_CREATE_INVALID_PROXY_HOST);
-
-    // check for missing proxyPort
-    Errors.checkArgumentExists(Util.exists(proxyPort),
-      ErrorCodes.ERR_CONN_CREATE_MISSING_PROXY_PORT);
-
-    // check for invalid proxyPort
-    Errors.checkArgumentValid(Util.isNumber(proxyPort),
-      ErrorCodes.ERR_CONN_CREATE_INVALID_PROXY_PORT);
-
-    if (Util.exists(noProxy)) {
-      // check for invalid noProxy
-      Errors.checkArgumentValid(Util.isString(noProxy),
-        ErrorCodes.ERR_CONN_CREATE_INVALID_NO_PROXY);
-    }
-
-    if (Util.exists(proxyUser) || Util.exists(proxyPassword)) {
-      // check for missing proxyUser
-      Errors.checkArgumentExists(Util.exists(proxyUser),
-        ErrorCodes.ERR_CONN_CREATE_MISSING_PROXY_USER);
-
-      // check for invalid proxyUser
-      Errors.checkArgumentValid(Util.isString(proxyUser),
-        ErrorCodes.ERR_CONN_CREATE_INVALID_PROXY_USER);
-
-      // check for missing proxyPassword
-      Errors.checkArgumentExists(Util.exists(proxyPassword),
-        ErrorCodes.ERR_CONN_CREATE_MISSING_PROXY_PASS);
-
-      // check for invalid proxyPassword
-      Errors.checkArgumentValid(Util.isString(proxyPassword),
-        ErrorCodes.ERR_CONN_CREATE_INVALID_PROXY_PASS);
-
-      proxy =
-      {
-        host: proxyHost,
-        port: proxyPort,
-        user: proxyUser,
-        password: proxyPassword,
-        protocol: proxyProtocol,
-        noProxy: noProxy
-      };
-    } else {
-      proxy =
-      {
-        host: proxyHost,
-        port: proxyPort,
-        protocol: proxyProtocol,
-        noProxy: noProxy
-      };
-    }
+    proxy =
+    {
+      host: proxyHost,
+      port: proxyPort,
+      user: proxyUser,
+      password: proxyPassword,
+      protocol: proxyProtocol,
+      noProxy: noProxy
```

Comment:
should we put a validator on this as well, similarly to proxy username and proxy password ? (delete if not supplied)

Reply:
Here it is: https://github.com/snowflakedb/snowflake-connector-nodejs/pull/938/files#diff-b74ee385f322c90ac5e644528cf9202501632d6350ad75c154e2c989a81e63b2R224